src/ElmFormat/Render/Box.hs

```diff
@@ -2121,6 +2065,87 @@ formatComment comment =
+    let
+        opening =
+          case
+            ( multilineSubject
+            , formatCommentedExpression elmVersion importInfo SyntaxSeparated subject
+            )
+          of
+              (AST.JoinAll, SingleLine subject') ->
+                  line $ row [ keyword "case" , space , subject' , space , keyword "of" ]
+              (_, subject') ->
+                  stack1
+                      [ line $ keyword "case"
+                      , indent subject'
+                      , line $ keyword "of"
+                      ]
+
+        (multilineBranches, branches') =
+            List.mapAccumR
+            (\mline (AST.Commented prePat pat postPat, (preRes, res)) ->
+                let
+                    prePat' =
+                        Maybe.maybeToList $ formatComments prePat
+                    pat' =
+                        formatPattern elmVersion False pat
+                            |> negativeCasePatternWorkaround pat
+                    postPat' =
+                        Maybe.maybeToList $ formatComments postPat
+                    preRes' =
+                        Maybe.maybeToList $ formatComments preRes
+                    res' =
+                        formatExpression elmVersion importInfo SyntaxSeparated res
+                    singles =
+                        allSingles $ concat [ prePat', [pat'], postPat', [line (keyword "->")], preRes', [res']]
+                in
+                case (multilineFirstBranch, singles, res') of
+                    (AST.JoinAll, Right _, SingleLine _) ->
+                        (mline, (singles, prePat', pat', postPat', preRes', res'))
+                    _ ->
+                        (AST.SplitAll, (singles, prePat', pat', postPat', preRes', res'))
+            )
+            multilineFirstBranch
+            branches
+
+        branch multiline' (singles, prePat, pat, postPat, preRes, res) =
+            case (multiline', singles, prePat, pat, postPat) of
+                (AST.JoinAll, Right singleLines, _, _, _) ->
+                    line $ row $ List.intersperse space singleLines
+                (_, _, [], SingleLine pat', [])  ->
+                    stack1
+                        [ line $ row [pat', space, keyword "->"]
+                        , indent $ stack1 $ preRes ++ [res]
+                        ]
+                (_, _, _, SingleLine pat', [])  ->
+                    stack1
+                        [ stack1 prePat
+                        , line $ row [pat', space, keyword "->"]
+                        , indent $ stack1 $ preRes ++ [res]
+                        ]
+                _  ->
+                    stack1
+                      [ stack1 $ prePat ++ [pat] ++ postPat
+                      , line $ keyword "->"
+                      , indent $ stack1 $ preRes ++ [res]
+                      ]
+    in
+    opening
+        |> andThen
+            (branches'
+                |> map (branch multilineBranches)
```

Comment:
Not necessary to change before merging, but FYI I'm trying to prefer `fmap` over `map`

Reply:
Fixed